golem-shard-manager/src/model.rs

```diff
@@ -79,11 +78,30 @@ impl Pod {
             ip: source_ip,
         };
 
-        let resolved = pod.address()?.map(|sa| sa.ip()).collect::<Vec<_>>();
-        if !resolved.contains(&source_ip) {
-            warn!("Host mismatch between registration message and resolved message source. Provided: {resolved:?}; source: {source_ip:?}");
+        match pod.address() {
+            Ok(resolved_addresses) => {
+                let resolved_addresses = resolved_addresses.map(|sa| sa.ip()).collect::<Vec<_>>();
+
+                if !resolved_addresses.contains(&source_ip) {
+                    warn!(
+                        pod= ?pod,
```

Comment:
should not we rather use Display (instead of Debug) ?

Reply:
👍 https://github.com/golemcloud/golem/pull/690/commits/92d61924bd8af9d2652c3e59c04a432e3a3f23b4